sdcm/utils/gce_utils.py

```diff
@@ -234,3 +266,278 @@ def _get_log_entries(self, service, query, page_token=None):
         if page_token := ret.get("nextPageToken"):
             entries.extend(self._get_log_entries(service, query, page_token))
         return entries
+
+
+def wait_for_extended_operation(
+    operation: ExtendedOperation, verbose_name: str = "operation", timeout: int = 300
+) -> Any:
+    """
+    Waits for the extended (long-running) operation to complete.
+
+    If the operation is successful, it will return its result.
+    If the operation ends with an error, an exception will be raised.
+    If there were any warnings during the execution of the operation
+    they will be printed to sys.stderr.
+
+    Args:
+        operation: a long-running operation you want to wait on.
+        verbose_name: (optional) a more verbose name of the operation,
+            used only during error and warning reporting.
+        timeout: how long (in seconds) to wait for operation to finish.
+            If None, wait indefinitely.
+
+    Returns:
+        Whatever the operation.result() returns.
+
+    Raises:
+        This method will raise the exception received from `operation.exception()`
+        or RuntimeError if there is no exception set, but there is an `error_code`
+        set for the `operation`.
+
+        In case of an operation taking longer than `timeout` seconds to complete,
+        a `concurrent.futures.TimeoutError` will be raised.
+    """
+    result = operation.result(timeout=timeout)
+
+    if operation.error_code:
+        LOGGER.debug("Error during %s: [Code: %s]: %s", verbose_name,
+                     operation.error_code, operation.error_message)
+        LOGGER.debug("Operation ID: %s", operation.name)
+        raise operation.exception() or RuntimeError(operation.error_message)
+
+    if operation.warnings:
+        LOGGER.debug("Warnings during %s:", verbose_name)
+        for warning in operation.warnings:
+            LOGGER.debug(" - %s: %s", warning.code, warning.message)
+
+    return result
+
+
+def disk_from_image(  # pylint: disable=too-many-arguments
+    disk_type: str,
+    boot: bool,
+    disk_size_gb: int = None,
+    source_image: str = None,
+    auto_delete: bool = True,
+    device_name: str = None,
+    type_: str = None,
```

Comment:
in cases it accepts specified set of strings, better use `Literal["PERSISTENT", "SCRATCH"]`

Reply:
added the typing